### DISCUSSION TITLE
Parcel v2 + microbundle, same project, two different build.

### DISCUSSION THREAD

**Discussion by nkint** [parcel-bundler/parcel#5039] [General] [answered]
Hi! 

I have project for a React component and I'd like to have two builds: one for a website/demo/doc that show the component, one for the component itself as a library.

The react component is in `src` and the doc page is in `src-doc`.

The project is written in typescript, and to bundle it I'm using the rollup powered utility called [microbundle](https://github.com/developit/microbundle) and a ts definition bundler called [dts-bundle-generator](https://github.com/timocov/dts-bundle-generator).

So that I have this package.json (I omitted unnecessary fields):

```json
{
	"scripts": {
		"build-doc": "parcel build src-doc/index.html ",
		"build-lib": "microbundle --no-compress && dts-bundle-generator ./src/index.ts",
		"start": "parcel src-doc/index.html",
	},
	"source": "src/index.ts",
	"main": "build/index.js",
	"types": "build/src/index.d.ts",
}
```

The fields `source`, `main` and `types` in the `package.json` are required by microbundle. If I leave them here parcel v2 says:

```
🚨 Build failed.
@parcel/namer-default: Target "main" declares an output file path of "build/index.js" which does not match the compiled bundle type "html".

> 27 |   "main": "build/index.js",
>    |           ^^^^^^^^^^^^^^^^ Did you mean "build/index.html"?
```

Right now I'm thinking of a small script that adds `main` and `types` field in the `package.json` just before running microbundle, and remove them after it. It is an ugly solution by the way.

Do you have any advice?

**Comment by mischnic** [comment, marked as answer]
Run `parcel build src-doc/index.html --target default` instead to ignore these main fields in package.json

**Comment by nkint** [comment]
wow, really thanks for fast reply 💥⚡️, you rocks!

@mischnic 

I get:

```
🚨 Build failed.
@parcel/core: Could not find target with name "default"
```

**Reply by nkint** [reply]
@mischnic sorry, my fault, I also need to add:

```json
"targets": {
	"default": {}
}
```

to `package.json`, as from [this doc page](https://v2.parceljs.org/getting-started/migration/#--target)
